Sming/Libraries/OtaUpgradeMqtt/src/include/OtaUpgrade/Mqtt/RbootPayloadParser.h

```diff
@@ -0,0 +1,43 @@
+/****
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
+ * Created 2015 by Skurydin Alexey
+ * http://github.com/SmingHub/Sming
+ * All files of the Sming Core are provided under the LGPL v3 license.
+ *
+ * PayloadParser.h
+ *
+ *  Created: 2021 - Slavey Karadzhov <slav@attachix.com>
+ *
+ ****/
+
+#pragma once
+
+#include "PayloadParser.h"
+#include <Data/Stream/RbootOutputStream.h>
+
+namespace OtaUpgrade
+{
+namespace Mqtt
+{
+/**
+ * @brief This parser allows the processing of firmware data that is directly stored
+ * 		  to the flash memory using RbootOutputStream.
+ */
+class RbootPayloadParser : public PayloadParser
+{
+public:
+	RbootPayloadParser(const Storage::Partition& part, size_t currentVersion)
```

Comment:
Technically this doesn't need to be a const&, and I'm wondering if it's misleading since we'll be writing to the partition. Might be clearer just writing `Storage::Partition part`.